### DISCUSSION TITLE
luv library changes

### DISCUSSION THREAD

**Discussion by T-727** [folke/lua-dev.nvim#135] [General]
Hi!
I recommend switching to the included `luv` library in `lua-language-server` a little while back #127.

And a few days ago a new `addon` system was released to replace the included libraries, in order to cut bloat.
And so the `3rd-party` system is deprecated and will be removed after some time.

I looked at how addons work and it seems the [intended way](https://github.com/LuaLS/lua-language-server/wiki/Addons#installing-addons) to use them for non-vscode users is cloning and adding to `workspace.userThirdParty`, for neodev purposes that either involves a workflow job or submodules as far as I'm aware.
For submodules unfortunately the [addons repo](https://github.com/LuaLS/LLS-Addons/tree/main/addons) does not ship `luv` as a separate library anymore, it is included as part of the [`luvit`](https://github.com/Bilal2453/luvit-meta/blob/0b3e46bc6fe412d94091314400fee67e76b2748c/library/uv.lua) addon instead ([relevant discussion](https://github.com/LuaLS/lua-language-server/discussions/1927)), so in order to add it as a submodule we either include the whole addon and be fine with the extra files or use workarounds such as [this](https://stackoverflow.com/questions/7597748/linking-a-single-file-from-another-git-repository#7597867) to include the single file that annotates `uv`.

If none of these are ideal we could also ask to get it separated in the discussion or make an issue.

**Comment by folke** [comment]
Thanks for letting me know!

I've just updated the build to include the single uv.lua file in the neodev types.
It will be updated every hour as part of the build.

https://github.com/folke/neodev.nvim/commit/f06c11344f76fadf2cd497b9490125dfc02946cb
